### PR TITLE
Feature/template images

### PR DESCRIPTION
**Pull request by EKuliiev**
Proposal:
https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0062-template-images.md

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Automated tests from: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1861

### Summary
Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0062-template-images.md

The API change on the mobile and HMI is the following:
<struct name="Image">
 <param name="value" minlength="0" maxlength="65535" type="String"></param>
 <param name="imageType" type="ImageType"></param>
 <param name="isTemplate" type="Boolean" mandatory="false"></param> 
</struct>

Whenever core receives an RPC from the app which incldues an image it just has to make sure the .isTemplate parameter makes its way to the HMI.

### Tasks Remaining:
- [ ] [Unit tests]

### CLA
- I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/commands/mobile/add_command_request.cc [hunk 1]
```diff
@@ -95,31 +96,28 @@ void AddCommandRequest::Run() {
...
+  if (app->FindCommand(msg_params[strings::cmd_id].asUInt())) {
     LOG4CXX_ERROR(logger_, "INVALID_ID");
     SendResponse(false, mobile_apis::Result::INVALID_ID);
     return;
   }
 
   bool data_exist = false;
 
-  if ((*message_)[strings::msg_params].keyExists(strings::menu_params)) {
+  if (msg_params.keyExists(strings::menu_params)) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/add_command_request.cc]: @EKuliiev please extract refactoring changes to separate commit

Comment (EKuliiev) [reply src/components/application_manager/src/commands/mobile/add_command_request.cc]: agree. done.

File: src/components/application_manager/src/resumption/resumption_data_db.cc [hunk 2]
```diff
@@ -1717,6 +1719,7 @@ bool ResumptionDataDB::ExecInsertImage(
     if (result) {
       query.Bind(0, image[strings::image_type].asInt());
       query.Bind(1, image[strings::value].asString());
+      query.Bind(2, image[strings::is_template].asBool());
```
[the remark this answers is not in the document]
Comment (EKuliiev) [reply src/components/application_manager/src/resumption/resumption_data_db.cc]: optional.

[the remark this answers is not in the document]
Comment (EKuliiev) [reply src/components/application_manager/src/resumption/resumption_data_db.cc]: I'll fix it

File: src/components/application_manager/src/resumption/resumption_data_db.cc [hunk 3]
```diff
@@ -1328,9 +1328,13 @@ bool ResumptionDataDB::SelectImageData(
   }
   /* Position of data in "select_image" :
      field "imageType" from table "image" = 0
-     field "value" from table "image" = 1*/
+     field "value" from table "image" = 1
+     optional field "isTemplate" from table "image" = 2 */
   image[strings::image_type] = select_image.GetInteger(0);
   image[strings::value] = select_image.GetString(1);
+  if (!select_image.IsNull(2)) {
```
Comment (EKuliiev) [inline src/components/application_manager/src/resumption/resumption_data_db.cc]: isTemplate is optional field.
@LuxoftAKutsan, I rewrite last commit
For now it check if field exist.
